test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/Util.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.XPlat.FuncTest
+{
+    internal class Util
+    {
+        private static string DotnetCliBinary { get; set; }
+
+        private static string DotnetCliExe { get; set; }
+
+        private static string XplatDll { get; set; }
+
+        private static string XplatDllShell { get; set; }
+
+        static Util()
+        {
+            DotnetCliBinary = @"dotnet";
+            DotnetCliExe = @"dotnet.exe";
+            XplatDll = @"NuGet.Core\NuGet.CommandLine.XPlat\bin\release\netcoreapp1.0\NuGet.CommandLine.XPlat.dll";
+            XplatDllShell = @"NuGet.Core/NuGet.CommandLine.XPlat/bin/release/netcoreapp1.0/NuGet.CommandLine.XPlat.dll";
+        }
+
+        /// <summary>
+        /// Provides the path to dotnet cli on the test machine.
+        /// It traverses in the directory tree going one step up at a time and looks for cli folder.
+        /// </summary>
+        /// <returns>
+        /// <code>String</code> containing the path to the dotnet cli within the local repository.
+        /// Can return <code>null</code> if no cli directory or dotnet cli is found, in which case the tests can fail.
+        /// </returns>
+        public static string GetDotnetCli()
+        {
+            var currentDirInfo = new DirectoryInfo(Directory.GetCurrentDirectory());
```

Comment:
Can Xunit or something else tell you the location of the test assembly and use that as a starting point? `Directory.GetCurrentDirectory` is just another static that can be modified by another test.
See:
https://docs.microsoft.com/en-us/dotnet/core/api/system.reflection.assembly#System_Reflection_Assembly_Location